File: cs/Minsk/CodeAnalysis/Binding/BoundUnaryExpression.cs
namespace Minsk.CodeAnalysis.Binding;

internal sealed class BoundUnaryExpression : BoundExpression
{
    public BoundUnaryExpression(BoundUnaryOperator op, BoundExpression operand)
    {
        Op = op;
        Operand = operand;
    }

    public BoundUnaryOperator Op { get; }
    public BoundExpression Operand { get; }
    public override Type Type => Op.ResultType;
    public override BoundNodeKind Kind => BoundNodeKind.UnaryExpression;
}

File: kt/Minsk/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/binding/BoundUnaryExpression.kt
package dev.phytolizer.minsk.analysis.binding

import java.lang.reflect.Type

internal class BoundUnaryExpression(val op: BoundUnaryOperator, val operand: BoundExpression) :
    BoundExpression() {

    override val kind: BoundNodeKind
        get() = BoundNodeKind.UnaryExpression
    override val type: Type
        get() = op.resultType
}
